### PR TITLE
Cache ROP gadgets

### PR DESCRIPTION
**Pull request by Darredevil**

### CODE REVIEW

**Comment by Maijin** [comment]
#1708

File: libr/core/cmd_search.c [hunk 1]
```diff
@@ -886,6 +886,7 @@ static void print_rop (RCore *core, RList *hitlist, char mode, bool *json_first)
 	RAsmOp asmop;
 	int colorize = r_config_get_i (core->config, "scr.color");
 	int rop_comments = r_config_get_i (core->config, "rop.comments");
+	int esil = r_config_get_i (core->config, "asm.esil");
```
Comment (radare) [inline libr/core/cmd_search.c]: this is bool, as well as the rop_comments and colorize

File: libr/core/cmd_search.c [hunk 2]
```diff
@@ -938,6 +937,13 @@ static void print_rop (RCore *core, RList *hitlist, char mode, bool *json_first)
 			}
 			free (buf);
 		}
+		if (esil && hit) {
+			r_cons_printf ("Gadget size: %d\n", size);
+			char input[80] = { 0 };
+			sprintf(input, "rop/0x%08"PFMT64x"=%d", ((RCoreAsmHit *)hitlist->head->data)->addr, size);
+
```
Comment (radare) [inline libr/core/cmd_search.c]: empty line

File: libr/core/cmd_search.c [hunk 3]
```diff
@@ -938,6 +937,13 @@ static void print_rop (RCore *core, RList *hitlist, char mode, bool *json_first)
 			}
 			free (buf);
 		}
+		if (esil && hit) {
+			r_cons_printf ("Gadget size: %d\n", size);
+			char input[80] = { 0 };
+			sprintf(input, "rop/0x%08"PFMT64x"=%d", ((RCoreAsmHit *)hitlist->head->data)->addr, size);
```
Comment (radare) [inline libr/core/cmd_search.c]: missing space before (

File: libr/core/cmd_search.c [hunk 4]
```diff
@@ -938,6 +937,13 @@ static void print_rop (RCore *core, RList *hitlist, char mode, bool *json_first)
 			}
 			free (buf);
 		}
+		if (esil && hit) {
+			r_cons_printf ("Gadget size: %d\n", size);
+			char input[80] = { 0 };
```
Comment (radare) [inline libr/core/cmd_search.c]: define variables at the begining of function frames. and no need to initialize it

File: libr/core/cmd_search.c (hunk 3 above)
Comment (radare) [inline libr/core/cmd_search.c]: use snprintf to avoid overflows

Comment (radare) [inline libr/core/cmd_search.c]: even if you know it fits

**Comment by radare** [comment]
please update the pr asap

**Comment by radare** [comment]
ping

**Comment by radare** [comment]
Merged and fixed all my comments. Thanks!

**Pull request closed without merging**